Source/WebCore/rendering/CSSFilterRenderer.cpp

```diff
@@ -48,68 +48,68 @@ namespace WebCore {
 
     return filterRenderer;
 }
 
-
-RefPtr<CSSFilterRenderer> CSSFilterRenderer::create(RenderElement& renderer, const Style::Filter& filter, OptionSet<FilterRenderingMode> preferredFilterRenderingModes, const FloatSize& filterScale, const FloatRect& targetBoundingBox, const GraphicsContext& destinationContext)
-{
-    return createGeneric(renderer, filter, preferredFilterRenderingModes, filterScale, targetBoundingBox, destinationContext);
-}
-
-RefPtr<CSSFilterRenderer> CSSFilterRenderer::create(RenderElement& renderer, const FilterOperations& operations, OptionSet<FilterRenderingMode> preferredFilterRenderingModes, const FloatSize& filterScale, const FloatRect& targetBoundingBox, const GraphicsContext& destinationContext)
+RefPtr<CSSFilterRenderer> CSSFilterRenderer::create(RenderElement& renderer, const Style::Filter& filter, const FilterInputs& inputs, const GraphicsContext& destinationContext)
 {
-    return createGeneric(renderer, operations, preferredFilterRenderingModes, filterScale, targetBoundingBox, destinationContext);
+    return createGeneric(renderer, filter, inputs, destinationContext);
 }
 
-Ref<CSSFilterRenderer> CSSFilterRenderer::create(Vector<Ref<FilterFunction>>&& functions)
+RefPtr<CSSFilterRenderer> CSSFilterRenderer::create(RenderElement& renderer, const FilterOperations& operations, const FilterInputs& inputs, const GraphicsContext& destinationContext)
 {
-    return adoptRef(*new CSSFilterRenderer(WTFMove(functions)));
+    return createGeneric(renderer, operations, inputs, destinationContext);
 }
 
-Ref<CSSFilterRenderer> CSSFilterRenderer::create(Vector<Ref<FilterFunction>>&& functions, OptionSet<FilterRenderingMode> filterRenderingModes, const FloatSize& filterScale, const FloatRect& filterRegion)
+Ref<CSSFilterRenderer> CSSFilterRenderer::create(Vector<Ref<FilterFunction>>&& functions, const FilterInputs& inputs)
 {
-    Ref filter = adoptRef(*new CSSFilterRenderer(WTFMove(functions), filterScale, filterRegion));
+    Ref filter = adoptRef(*new CSSFilterRenderer(WTFMove(functions), inputs));
     // Setting filter rendering modes cannot be moved to the constructor because it ends up
     // calling supportedFilterRenderingModes() which is a virtual function.
-    filter->setFilterRenderingModes(filterRenderingModes);
+    filter->setFilterRenderingModes(inputs.renderingModes);
     return filter;
 }
 
-CSSFilterRenderer::CSSFilterRenderer(const FloatSize& filterScale, bool hasFilterThatMovesPixels, bool hasFilterThatShouldBeRestrictedBySecurityOrigin)
-    : Filter(Filter::Type::CSSFilterRenderer, filterScale)
+CSSFilterRenderer::CSSFilterRenderer(const FilterInputs& inputs, bool hasFilterThatMovesPixels, bool hasFilterThatShouldBeRestrictedBySecurityOrigin)
+    : Filter(Filter::Type::CSSFilterRenderer, inputs.scale, inputs.filterRegion)
```

Comment:
I think the constructor of `Filter` should take `FilterInputs` as an argument and store it in a member. So we do not have these members in Filter.h
```
private:
    OptionSet<FilterRenderingMode> m_filterRenderingModes { FilterRenderingMode::Software };
    FloatSize m_filterScale;
    FloatRect m_filterRegion;
};
```
But we have `FilterInputs` instead.